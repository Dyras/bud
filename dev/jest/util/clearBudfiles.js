const {noop} = require('lodash')
const rimraf = require('rimraf')

module.exports = async () => {
  return rimraf(
    `${process.cwd()}/examples/*/.budfiles`,
    {
      glob: true,
    },
    noop,
  )
}
