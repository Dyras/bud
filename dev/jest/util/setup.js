/* eslint-disable no-console */
// @ts-check

const chalk = require('chalk')
const execa = require('execa')

const clearArtifacts = require('./clearArtifacts')
const paths = require('./paths')

const task = async (cmd, paths) => {
  console.log(chalk.blue`\nyarn bud ${cmd}\n`)

  await Promise.all(
    paths.map(async ex => {
      const task = execa.command(`yarn bud ${cmd}`, {
        cwd: ex.cwd,
      })
      task.stdout.pipe(process.stdout)
      await task
    }),
  )
}

module.exports = async () => {
  await clearArtifacts()

  const examples = await paths()

  await task('build --no-dashboard', examples)
}
