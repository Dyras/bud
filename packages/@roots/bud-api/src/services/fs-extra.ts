import {fs} from '@roots/bud-support'

export const ensureDirSync = fs.ensureDirSync
export const pathExistsSync = fs.pathExistsSync
