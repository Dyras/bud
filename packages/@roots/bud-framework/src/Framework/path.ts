import {join} from 'path'

import {Framework} from '..'

interface path {
  (
    this: Framework,
    key: keyof Framework.Locations,
    ...path: string[]
  ): string
}

function path(
  key: keyof Framework.Locations,
  ...path: string[]
): string {
  return join(
    ...[
      key !== 'project'
        ? this.hooks.filter('location/project')
        : false,
      this.hooks.filter(`location/${key}`),
      ...(path ?? []),
    ].filter(Boolean),
  )
}

export {path}
