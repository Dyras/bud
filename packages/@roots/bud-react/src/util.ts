/**
 * @module @roots/bud-react
 */

import type {Configuration} from 'webpack'

/**
 * @function addRefresh
 */
function addRefresh(entries, [name, assets]) {
  return {
    ...(entries ?? {}),
    [name]: {
      ...assets,
      import: [
        'react-refresh/runtime',
        ...(assets.import ?? []),
      ],
    },
  }
}

/**
 * @function reducer
 */
function reducer(entry: Configuration['entry']) {
  return entry
    ? Object.entries(entry).reduce(addRefresh, entry)
    : {}
}

/**
 * @exports default
 */
export default reducer
