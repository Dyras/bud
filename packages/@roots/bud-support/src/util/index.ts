export {notify} from './notify'
export {dump} from './dump'
export {format} from './format'
export {processHandler} from './processHandler'

import InterpolateHtmlPlugin from './InterpolateHtmlPlugin'
export {InterpolateHtmlPlugin}
