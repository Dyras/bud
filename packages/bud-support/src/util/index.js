"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterpolateHtmlPlugin = exports.formatWebpackMessages = exports.eslintFormatter = exports.processHandler = exports.format = exports.dump = exports.notify = void 0;
var notify_1 = require("./notify");
Object.defineProperty(exports, "notify", { enumerable: true, get: function () { return notify_1.notify; } });
var dump_1 = require("./dump");
Object.defineProperty(exports, "dump", { enumerable: true, get: function () { return dump_1.dump; } });
var format_1 = require("./format");
Object.defineProperty(exports, "format", { enumerable: true, get: function () { return format_1.format; } });
var processHandler_1 = require("./processHandler");
Object.defineProperty(exports, "processHandler", { enumerable: true, get: function () { return processHandler_1.processHandler; } });
const eslintFormatter_1 = __importDefault(require("./eslintFormatter"));
exports.eslintFormatter = eslintFormatter_1.default;
const InterpolateHtmlPlugin_1 = __importDefault(require("./InterpolateHtmlPlugin"));
exports.InterpolateHtmlPlugin = InterpolateHtmlPlugin_1.default;
const formatWebpackMessages_1 = __importDefault(require("./formatWebpackMessages"));
exports.formatWebpackMessages = formatWebpackMessages_1.default;
//# sourceMappingURL=index.js.map