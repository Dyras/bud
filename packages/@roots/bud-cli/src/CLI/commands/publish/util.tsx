import {chalk, globby} from '@roots/bud-support'
import {dirname} from 'path'

/**
 * Publishable matches (glob pattern)
 */
export const publishables = [
  ...globby
    .sync([
      '**/node_modules/bud-*/publish/*',
      '**/node_modules/*/bud-*/publish/*',
    ])
    .filter(
      (value, index, self) => self.indexOf(value) === index,
    ),
]

export const formatted = publishables.reduce((a, file) => {
  const ext = dirname(dirname(file)).replace('node_modules/', '')

  const template = file
    .replace(`${ext}/publish/`, '')
    .replace('node_modules/', '')

  return `${a} \n${chalk.blue(ext)} ${chalk.green(template)}`
}, ``)
