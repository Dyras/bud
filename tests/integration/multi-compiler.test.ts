import execa from 'execa'
import {readFile, readJson} from 'fs-extra'
import {join} from 'path'

import {Assets} from '../util/integration'

interface CompilerArtifacts {
  assets: Assets
  manifest: any
}

const cwd = `${process.cwd()}/examples/multi-compiler`

jest.setTimeout(60000)

describe('multi-compiler', () => {
  const parent: CompilerArtifacts = {
    assets: {},
    manifest: {},
  }

  const plugin: CompilerArtifacts = {
    assets: {},
    manifest: {},
  }

  const theme: CompilerArtifacts = {
    assets: {},
    manifest: {},
  }

  beforeAll(async () => {
    await execa('yarn', ['bud', 'extensions:install'], {
      cwd,
    })
    await execa('yarn', ['bud', `build:production`, '--ci'], {
      cwd,
    })

    parent.manifest = await readJson(
      `${cwd}/dist/global/manifest.json`,
    )
    parent.assets = await Object.entries(parent.manifest).reduce(
      async (promised: Promise<any>, [name, path]) => {
        const assets = await promised
        const buffer = await readFile(
          `${cwd}/dist/global/${path}`,
          'utf8',
        )

        return {
          ...assets,
          [name]: buffer.toString(),
        }
      },
      Promise.resolve(),
    )

    plugin.manifest = await readJson(
      `${cwd}/dist/plugin/manifest.json`,
    )
    plugin.assets = await Object.entries(plugin.manifest).reduce(
      async (promised: Promise<any>, [name, path]) => {
        const assets = await promised
        const buffer = await readFile(
          `${cwd}/dist/plugin/${path}`,
          'utf8',
        )

        return {
          ...assets,
          [name]: buffer.toString(),
        }
      },
      Promise.resolve(),
    )

    theme.manifest = await readJson(
      `${cwd}/dist/theme/manifest.json`,
    )
    theme.assets = await Object.entries(theme.manifest).reduce(
      async (promised: Promise<any>, [name, path]) => {
        const assets = await promised
        const buffer = await readFile(
          `${cwd}/dist/theme/${path}`,
          'utf8',
        )

        return {
          ...assets,
          [name]: buffer.toString(),
        }
      },
      Promise.resolve(),
    )

    return Promise.resolve()
  })

  describe('global.js', () => {
    it('has contents', () => {
      expect(parent.assets['global.js'].length).toBeGreaterThan(
        10,
      )
    })

    it('is transpiled', () => {
      expect(
        parent.assets['global.js'].includes('import'),
      ).toBeFalsy()
    })
  })

  describe('theme.js', () => {
    it('has contents', () => {
      expect(theme.assets['theme.js'].length).toBeGreaterThan(10)
    })

    it('is transpiled', () => {
      expect(
        theme.assets['theme.js'].includes('import'),
      ).toBeFalsy()
    })
  })

  describe('plugin.js', () => {
    it('has contents', () => {
      expect(plugin.assets['plugin.js'].length).toBeGreaterThan(
        10,
      )
    })

    it('is transpiled', () => {
      expect(
        plugin.assets['plugin.js'].includes('import'),
      ).toBeFalsy()
    })
  })

  it('module map matches snapshot', async () => {
    const artifact = await readJson(
      join(
        process.cwd(),
        'examples/multi-compiler/.budfiles/bud-modules.json',
      ),
    )

    expect(artifact.chunks).toMatchSnapshot({
      byName: {
        global: expect.any(Number),
      },
      bySource: {
        '0 global': expect.any(Number),
      },
      usedIds: [expect.any(Number)],
    })
  })
})
