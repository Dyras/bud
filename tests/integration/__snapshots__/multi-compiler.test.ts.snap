// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multi-compiler module map matches snapshot 1`] = `
Object {
  "byName": Object {
    "global": Any<Number>,
  },
  "bySource": Object {
    "0 global": Any<Number>,
  },
  "usedIds": Array [
    Any<Number>,
  ],
}
`;
